src/actions/user.js

```diff
@@ -27,25 +27,27 @@ export const updateUserSnippetsFailed = () => ({
   type: UPDATE_USER_SNIPPETS_FAILED,
 });
 
-export const updateUserSnippets = () => {
-  return (dispatch) => {
+export const updateUserSnippets = () => (dispatch) => {
     // Load requisite cookies, return Promise if they aren't present
-    const token    = cookie.load('token');
-    const username = cookie.load('username');
+  const token = cookie.load('token');
+  const username = cookie.load('username');
 
     // Fetch the user's snippet meta data and save it
-    const headers = {
-      Accept: 'application/json',
-      Authorization: `token ${token}`,
-    };
-    dispatch(updateUserSnippetsStarted());
-    return axios.get(makeSaveEndpointUrl(username), { headers })
+  const headers = {
+    Accept: 'application/json',
+    Authorization: `token ${token}`,
+  };
+  dispatch(updateUserSnippetsStarted());
+  return axios.get(makeSaveEndpointUrl(username), { headers })
       .then(
-        res => {
+        (res) => {
+          // TODO: Don't execute the request if there is no username
+          if (typeof res.data === 'string') {
```

Comment:
Can you drop a comment here explaining what has happened if `res.data` is a `string`? Right now it is not at all clear why we would return if that happens.